x-pack/test/functional/apps/spaces/spaces_selection.ts

```diff
@@ -83,19 +83,16 @@ export default function spaceSelectorFunctonalTests({ getService, getPageObjects
       });
 
       after(async () => {
+        // No need to remove the same sample data in both spaces, the index
+        // data will be removed in the first call. By feature limitation,
+        // the created saved objects in the second space will be broken but removed
+        // when we call esArchiver.unload('spaces').
         await PageObjects.common.navigateToApp('home', {
           appConfig: {
             hash: sampleDataHash,
           },
         });
         await PageObjects.home.removeSampleDataSet('logs');
-        await PageObjects.common.navigateToApp('home', {
-          appConfig: {
-            hash: sampleDataHash,
-            pathname: `/s/${spaceId}${homePath}`,
-          },
-        });
-        await PageObjects.home.removeSampleDataSet('logs');
```

Comment:
Removing this for the time being makes sense to me, especially if `esArchiver.unload()` is doing the cleanup already.

In the future when #24263 is resolved, we can probably remove all of the cleanup (except esArchiver), and then create a separate test for this particular edge case.